plutus-tx-plugin/test/AsData/Budget/9.6/destructSum-manual.uplc.golden

```diff
@@ -3,62 +3,56 @@
    (\d ->
       (\cse ->
          (\cse ->
-            (\cse ->
-               case
-                 cse
-                 [ (case
-                      cse
-                      [ (case
-                           (equalsInteger 2 cse)
-                           [ ((\cse -> case cse [cse]) error)
-                           , ((\args ->
-                                 (\y ->
-                                    (\`$mInts` ->
-                                       `$mInts`
-                                         (force headList args)
-                                         (\x y z w ->
-                                            `$mInts`
-                                              y
-                                              (\x y z w ->
-                                                 constrData
-                                                   0
+            case
+              (equalsInteger 0 cse)
+              [ (case
+                   (equalsInteger 1 cse)
+                   [ (case
+                        (equalsInteger 2 cse)
+                        [ ((\cse -> case cse [cse]) error)
```

Comment:
We should have an optimization that rewrites this (and anything else that forces `error`) to `error`.